# ChIP-seq and ATAC-seq analysis (23 - 24 October 2019) 
### 23rd - 24th October 2019
#### Bioinformatics Training Room, Craik-Marshall Building, Downing Site, University of Cambridge

## Outline

The primary aim of this course is to familiarise participants with the analysis
of ChIP-seq and ATAC-seq data and provide hands-on training on the latest
analytical approaches.

The course starts with an introduction to ChIP-seq experiments for the
detection of genome-wide DNA binding sites of transcription factors and other
proteins. We first show data quality control and basic analytical steps such as
alignment, peak calling and motif analysis, followed by practical examples on
how to work with biological replicates and fundamental quality metrics for
ChIP-seq datasets. On the second day, we then focus on the analysis of
differential binding, comparing between different samples. We will also give an
introduction to ATAC-seq data analysis for the detection of regions of open
chromatin. 

## Instructors

* Sandra Cortijo - Locke Group, Sainsbury Laboratory, Cambridge University
* Sergio Martinez Cuesta - Balasubramanian Group, Cancer Research UK Cambridge Institute   
* Sankari Nagarajan - Carroll Group, Cancer Research UK Cambridge Institute  
* Ashley Sawle - Bioinformatics Core, Cancer Research UK Cambridge Institute 

## Timetable

### Day 1
09:30 - 10:30 	Introduction to ChIP-seq, data analysis and QC metrics  
> Lecture: [Introduction to high throughput sequencing](Lectures/HTS/HTSIntro.pdf)  
> Lecture: [Sequencing data formats and Quality Control](Lectures/QC/QC_and_Formats.pdf)  
10:30 - 12:30 	Practical: [Introduction to ChIP-seq, data analysis and QC metrics](Practicals/QC/quality-control.html)  
12:30 - 13:30 	Lunch (not provided)  
13:30 - 14:30 	Lecture: [ChIP-seq data analysis](Lectures/ChIPseq_Introduction/ChIP-seq_intro_slides.html)  
15:30 - 17:30 	Practical: [ChIP-seq data analysis](Practicals/ChIPseq/oct4-chip-seq.html)  

### Day 2 	

09:30 - 10:00 	Lecture: [Recap and biological replicates](Lectures/IDR/20191024_recap_replicates.pdf)  
10:00 - 11:00 	Practical: [Working with biological replicates](Practicals/IDR/Replicate_IDR.html)  
11:00 - 11:15   Tea Break  
11:15 - 11:45 	Lecture: [Differential binding analysis](Lectures/Differential_Binding/Differential_Binding.html)  
11:45 - 13:00 	Practical: [Differential binding analysis](Practicals/Differential_Binding/DiffBind_practical.html)  
13:00 - 14:00 	Lunch (not provided)  
14:00 - 14:45 	Lecture: [Introduction to ATAC-seq data analysis](Lectures/ATAC/ATAC-seq.pdf)  
14:45 - 15:00 	Tea/coffee break  
15:00 - 17:00 	Practical: [ATAC-seq data analysis ](Practicals/ATACseq/ATACseq_tutorial.html)  

> ## Prerequisites
>
> * Basic understanding of high-throughput sequencing technologies  
> * Basic experience of command line UNIX. Sufficient UNIX experience might be
>   obtained from the following UNIX tutorial  
> * Basic knowledge of the R syntax. We recommend either attending Introduction
>   to R for biologists, or working through the materials of the now
>   discontinued An Introduction to Solving Biological Problems with R or Data
>   Carpentry in R before attending this course.  


## Source Materials for Practicals

The all of the lecture slides and other materials can be found in the course's [Github 
repository](https://github.com/cambiotraining/chipseq)


