<!DOCTYPE html>
<html>
  <head>
      <meta charset="utf-8" />
      <title>20190503_replicates</title>
      <style>.markdown-preview:not([data-use-github-style]) { padding: 2em; font-size: 1.2em; color: rgb(85, 85, 85); background-color: white; overflow: auto; }
.markdown-preview:not([data-use-github-style]) > :first-child { margin-top: 0px; }
.markdown-preview:not([data-use-github-style]) h1, .markdown-preview:not([data-use-github-style]) h2, .markdown-preview:not([data-use-github-style]) h3, .markdown-preview:not([data-use-github-style]) h4, .markdown-preview:not([data-use-github-style]) h5, .markdown-preview:not([data-use-github-style]) h6 { line-height: 1.2; margin-top: 1.5em; margin-bottom: 0.5em; color: rgb(3, 3, 3); }
.markdown-preview:not([data-use-github-style]) h1 { font-size: 2.4em; font-weight: 300; }
.markdown-preview:not([data-use-github-style]) h2 { font-size: 1.8em; font-weight: 400; }
.markdown-preview:not([data-use-github-style]) h3 { font-size: 1.5em; font-weight: 500; }
.markdown-preview:not([data-use-github-style]) h4 { font-size: 1.2em; font-weight: 600; }
.markdown-preview:not([data-use-github-style]) h5 { font-size: 1.1em; font-weight: 600; }
.markdown-preview:not([data-use-github-style]) h6 { font-size: 1em; font-weight: 600; }
.markdown-preview:not([data-use-github-style]) strong { color: rgb(3, 3, 3); }
.markdown-preview:not([data-use-github-style]) del { color: rgb(126, 126, 126); }
.markdown-preview:not([data-use-github-style]) a, .markdown-preview:not([data-use-github-style]) a code { color: black; }
.markdown-preview:not([data-use-github-style]) img { max-width: 100%; }
.markdown-preview:not([data-use-github-style]) > p { margin-top: 0px; margin-bottom: 1.5em; }
.markdown-preview:not([data-use-github-style]) > ul, .markdown-preview:not([data-use-github-style]) > ol { margin-bottom: 1.5em; }
.markdown-preview:not([data-use-github-style]) blockquote { margin: 1.5em 0px; font-size: inherit; color: rgb(126, 126, 126); border-color: rgb(214, 214, 214); border-width: 4px; }
.markdown-preview:not([data-use-github-style]) hr { margin: 3em 0px; border-top: 2px dashed rgb(214, 214, 214); background: none; }
.markdown-preview:not([data-use-github-style]) table { margin: 1.5em 0px; }
.markdown-preview:not([data-use-github-style]) th { color: rgb(3, 3, 3); }
.markdown-preview:not([data-use-github-style]) th, .markdown-preview:not([data-use-github-style]) td { padding: 0.66em 1em; border: 1px solid rgb(214, 214, 214); }
.markdown-preview:not([data-use-github-style]) code { color: rgb(3, 3, 3); background-color: rgb(240, 240, 240); }
.markdown-preview:not([data-use-github-style]) pre.editor-colors { margin: 1.5em 0px; padding: 1em; font-size: 0.92em; border-radius: 3px; background-color: rgb(245, 245, 245); }
.markdown-preview:not([data-use-github-style]) kbd { color: rgb(3, 3, 3); border-width: 1px 1px 2px; border-style: solid; border-color: rgb(214, 214, 214) rgb(214, 214, 214) rgb(199, 199, 199); border-image: initial; background-color: rgb(240, 240, 240); }
.markdown-preview[data-use-github-style] { font-family: "Helvetica Neue", Helvetica, "Segoe UI", Arial, freesans, sans-serif; line-height: 1.6; word-wrap: break-word; padding: 30px; font-size: 16px; color: rgb(51, 51, 51); background-color: rgb(255, 255, 255); overflow: scroll; }
.markdown-preview[data-use-github-style] > :first-child { margin-top: 0px !important; }
.markdown-preview[data-use-github-style] > :last-child { margin-bottom: 0px !important; }
.markdown-preview[data-use-github-style] a:not([href]) { color: inherit; text-decoration: none; }
.markdown-preview[data-use-github-style] .absent { color: rgb(204, 0, 0); }
.markdown-preview[data-use-github-style] .anchor { position: absolute; top: 0px; left: 0px; display: block; padding-right: 6px; padding-left: 30px; margin-left: -30px; }
.markdown-preview[data-use-github-style] .anchor:focus { outline: none; }
.markdown-preview[data-use-github-style] h1, .markdown-preview[data-use-github-style] h2, .markdown-preview[data-use-github-style] h3, .markdown-preview[data-use-github-style] h4, .markdown-preview[data-use-github-style] h5, .markdown-preview[data-use-github-style] h6 { position: relative; margin-top: 1em; margin-bottom: 16px; font-weight: bold; line-height: 1.4; }
.markdown-preview[data-use-github-style] h1 .octicon-link, .markdown-preview[data-use-github-style] h2 .octicon-link, .markdown-preview[data-use-github-style] h3 .octicon-link, .markdown-preview[data-use-github-style] h4 .octicon-link, .markdown-preview[data-use-github-style] h5 .octicon-link, .markdown-preview[data-use-github-style] h6 .octicon-link { display: none; color: rgb(0, 0, 0); vertical-align: middle; }
.markdown-preview[data-use-github-style] h1:hover .anchor, .markdown-preview[data-use-github-style] h2:hover .anchor, .markdown-preview[data-use-github-style] h3:hover .anchor, .markdown-preview[data-use-github-style] h4:hover .anchor, .markdown-preview[data-use-github-style] h5:hover .anchor, .markdown-preview[data-use-github-style] h6:hover .anchor { padding-left: 8px; margin-left: -30px; text-decoration: none; }
.markdown-preview[data-use-github-style] h1:hover .anchor .octicon-link, .markdown-preview[data-use-github-style] h2:hover .anchor .octicon-link, .markdown-preview[data-use-github-style] h3:hover .anchor .octicon-link, .markdown-preview[data-use-github-style] h4:hover .anchor .octicon-link, .markdown-preview[data-use-github-style] h5:hover .anchor .octicon-link, .markdown-preview[data-use-github-style] h6:hover .anchor .octicon-link { display: inline-block; }
.markdown-preview[data-use-github-style] h1 tt, .markdown-preview[data-use-github-style] h2 tt, .markdown-preview[data-use-github-style] h3 tt, .markdown-preview[data-use-github-style] h4 tt, .markdown-preview[data-use-github-style] h5 tt, .markdown-preview[data-use-github-style] h6 tt, .markdown-preview[data-use-github-style] h1 code, .markdown-preview[data-use-github-style] h2 code, .markdown-preview[data-use-github-style] h3 code, .markdown-preview[data-use-github-style] h4 code, .markdown-preview[data-use-github-style] h5 code, .markdown-preview[data-use-github-style] h6 code { font-size: inherit; }
.markdown-preview[data-use-github-style] h1 { padding-bottom: 0.3em; font-size: 2.25em; line-height: 1.2; border-bottom: 1px solid rgb(238, 238, 238); }
.markdown-preview[data-use-github-style] h1 .anchor { line-height: 1; }
.markdown-preview[data-use-github-style] h2 { padding-bottom: 0.3em; font-size: 1.75em; line-height: 1.225; border-bottom: 1px solid rgb(238, 238, 238); }
.markdown-preview[data-use-github-style] h2 .anchor { line-height: 1; }
.markdown-preview[data-use-github-style] h3 { font-size: 1.5em; line-height: 1.43; }
.markdown-preview[data-use-github-style] h3 .anchor { line-height: 1.2; }
.markdown-preview[data-use-github-style] h4 { font-size: 1.25em; }
.markdown-preview[data-use-github-style] h4 .anchor { line-height: 1.2; }
.markdown-preview[data-use-github-style] h5 { font-size: 1em; }
.markdown-preview[data-use-github-style] h5 .anchor { line-height: 1.1; }
.markdown-preview[data-use-github-style] h6 { font-size: 1em; color: rgb(119, 119, 119); }
.markdown-preview[data-use-github-style] h6 .anchor { line-height: 1.1; }
.markdown-preview[data-use-github-style] p, .markdown-preview[data-use-github-style] blockquote, .markdown-preview[data-use-github-style] ul, .markdown-preview[data-use-github-style] ol, .markdown-preview[data-use-github-style] dl, .markdown-preview[data-use-github-style] table, .markdown-preview[data-use-github-style] pre { margin-top: 0px; margin-bottom: 16px; }
.markdown-preview[data-use-github-style] hr { height: 4px; padding: 0px; margin: 16px 0px; background-color: rgb(231, 231, 231); border: 0px none; }
.markdown-preview[data-use-github-style] ul, .markdown-preview[data-use-github-style] ol { padding-left: 2em; }
.markdown-preview[data-use-github-style] ul.no-list, .markdown-preview[data-use-github-style] ol.no-list { padding: 0px; list-style-type: none; }
.markdown-preview[data-use-github-style] ul ul, .markdown-preview[data-use-github-style] ul ol, .markdown-preview[data-use-github-style] ol ol, .markdown-preview[data-use-github-style] ol ul { margin-top: 0px; margin-bottom: 0px; }
.markdown-preview[data-use-github-style] li > p { margin-top: 16px; }
.markdown-preview[data-use-github-style] dl { padding: 0px; }
.markdown-preview[data-use-github-style] dl dt { padding: 0px; margin-top: 16px; font-size: 1em; font-style: italic; font-weight: bold; }
.markdown-preview[data-use-github-style] dl dd { padding: 0px 16px; margin-bottom: 16px; }
.markdown-preview[data-use-github-style] blockquote { padding: 0px 15px; color: rgb(119, 119, 119); border-left: 4px solid rgb(221, 221, 221); }
.markdown-preview[data-use-github-style] blockquote > :first-child { margin-top: 0px; }
.markdown-preview[data-use-github-style] blockquote > :last-child { margin-bottom: 0px; }
.markdown-preview[data-use-github-style] table { display: block; width: 100%; overflow: auto; word-break: keep-all; }
.markdown-preview[data-use-github-style] table th { font-weight: bold; }
.markdown-preview[data-use-github-style] table th, .markdown-preview[data-use-github-style] table td { padding: 6px 13px; border: 1px solid rgb(221, 221, 221); }
.markdown-preview[data-use-github-style] table tr { background-color: rgb(255, 255, 255); border-top: 1px solid rgb(204, 204, 204); }
.markdown-preview[data-use-github-style] table tr:nth-child(2n) { background-color: rgb(248, 248, 248); }
.markdown-preview[data-use-github-style] img { max-width: 100%; box-sizing: border-box; }
.markdown-preview[data-use-github-style] .emoji { max-width: none; }
.markdown-preview[data-use-github-style] span.frame { display: block; overflow: hidden; }
.markdown-preview[data-use-github-style] span.frame > span { display: block; float: left; width: auto; padding: 7px; margin: 13px 0px 0px; overflow: hidden; border: 1px solid rgb(221, 221, 221); }
.markdown-preview[data-use-github-style] span.frame span img { display: block; float: left; }
.markdown-preview[data-use-github-style] span.frame span span { display: block; padding: 5px 0px 0px; clear: both; color: rgb(51, 51, 51); }
.markdown-preview[data-use-github-style] span.align-center { display: block; overflow: hidden; clear: both; }
.markdown-preview[data-use-github-style] span.align-center > span { display: block; margin: 13px auto 0px; overflow: hidden; text-align: center; }
.markdown-preview[data-use-github-style] span.align-center span img { margin: 0px auto; text-align: center; }
.markdown-preview[data-use-github-style] span.align-right { display: block; overflow: hidden; clear: both; }
.markdown-preview[data-use-github-style] span.align-right > span { display: block; margin: 13px 0px 0px; overflow: hidden; text-align: right; }
.markdown-preview[data-use-github-style] span.align-right span img { margin: 0px; text-align: right; }
.markdown-preview[data-use-github-style] span.float-left { display: block; float: left; margin-right: 13px; overflow: hidden; }
.markdown-preview[data-use-github-style] span.float-left span { margin: 13px 0px 0px; }
.markdown-preview[data-use-github-style] span.float-right { display: block; float: right; margin-left: 13px; overflow: hidden; }
.markdown-preview[data-use-github-style] span.float-right > span { display: block; margin: 13px auto 0px; overflow: hidden; text-align: right; }
.markdown-preview[data-use-github-style] code, .markdown-preview[data-use-github-style] tt { padding: 0.2em 0px; margin: 0px; font-size: 85%; background-color: rgba(0, 0, 0, 0.04); border-radius: 3px; }
.markdown-preview[data-use-github-style] code::before, .markdown-preview[data-use-github-style] tt::before, .markdown-preview[data-use-github-style] code::after, .markdown-preview[data-use-github-style] tt::after { letter-spacing: -0.2em; content: " "; }
.markdown-preview[data-use-github-style] code br, .markdown-preview[data-use-github-style] tt br { display: none; }
.markdown-preview[data-use-github-style] del code { text-decoration: inherit; }
.markdown-preview[data-use-github-style] pre > code { padding: 0px; margin: 0px; font-size: 100%; word-break: normal; white-space: pre; background: transparent; border: 0px; }
.markdown-preview[data-use-github-style] .highlight { margin-bottom: 16px; }
.markdown-preview[data-use-github-style] .highlight pre, .markdown-preview[data-use-github-style] pre { padding: 16px; overflow: auto; font-size: 85%; line-height: 1.45; background-color: rgb(247, 247, 247); border-radius: 3px; }
.markdown-preview[data-use-github-style] .highlight pre { margin-bottom: 0px; word-break: normal; }
.markdown-preview[data-use-github-style] pre { word-wrap: normal; }
.markdown-preview[data-use-github-style] pre code, .markdown-preview[data-use-github-style] pre tt { display: inline; max-width: initial; padding: 0px; margin: 0px; overflow: initial; line-height: inherit; word-wrap: normal; background-color: transparent; border: 0px; }
.markdown-preview[data-use-github-style] pre code::before, .markdown-preview[data-use-github-style] pre tt::before, .markdown-preview[data-use-github-style] pre code::after, .markdown-preview[data-use-github-style] pre tt::after { content: normal; }
.markdown-preview[data-use-github-style] kbd { display: inline-block; padding: 3px 5px; font-size: 11px; line-height: 10px; color: rgb(85, 85, 85); vertical-align: middle; background-color: rgb(252, 252, 252); border-width: 1px; border-style: solid; border-color: rgb(204, 204, 204) rgb(204, 204, 204) rgb(187, 187, 187); border-image: initial; border-radius: 3px; box-shadow: rgb(187, 187, 187) 0px -1px 0px inset; }
.markdown-preview[data-use-github-style] a { color: rgb(51, 122, 183); }
.markdown-preview[data-use-github-style] code { color: inherit; }
.markdown-preview[data-use-github-style] pre.editor-colors { padding: 0.8em 1em; margin-bottom: 1em; font-size: 0.85em; border-radius: 4px; overflow: auto; }
.markdown-preview pre.editor-colors { user-select: auto; }
.scrollbars-visible-always .markdown-preview pre.editor-colors .vertical-scrollbar, .scrollbars-visible-always .markdown-preview pre.editor-colors .horizontal-scrollbar { visibility: hidden; }
.scrollbars-visible-always .markdown-preview pre.editor-colors:hover .vertical-scrollbar, .scrollbars-visible-always .markdown-preview pre.editor-colors:hover .horizontal-scrollbar { visibility: visible; }
.markdown-preview .task-list-item-checkbox { position: absolute; margin: 0.25em 0px 0px -1.4em; }
.markdown-preview ul label { vertical-align: top; }
.bracket-matcher .region {
  border-bottom: 1px dotted lime;
  position: absolute;
}
.line-number.bracket-matcher.bracket-matcher {
  color: #555;
  background-color: rgba(255, 255, 134, 0.34);
}

pre.editor-colors {
  background-color: white;
  color: #555;
}
pre.editor-colors .invisible-character {
  color: rgba(85, 85, 85, 0.2);
}
pre.editor-colors .indent-guide {
  color: rgba(85, 85, 85, 0.2);
}
pre.editor-colors .wrap-guide {
  background-color: rgba(85, 85, 85, 0.2);
}
pre.editor-colors .gutter {
  color: #555;
  background: white;
}
pre.editor-colors .gutter .line-number.folded,
pre.editor-colors .gutter .line-number:after,
pre.editor-colors .fold-marker:after {
  color: #e87b00;
}
pre.editor-colors .invisible {
  color: #555;
}
pre.editor-colors .selection .region {
  background-color: #e1e1e1;
}
pre.editor-colors .bracket-matcher .region {
  background-color: #C9C9C9;
  opacity: .7;
  border-bottom: 0 none;
}
pre.editor-colors.is-focused .cursor {
  border-color: black;
}
pre.editor-colors.is-focused .selection .region {
  background-color: #afc4da;
}
pre.editor-colors.is-focused .line-number.cursor-line-no-selection,
pre.editor-colors.is-focused .line.cursor-line {
  background-color: rgba(255, 255, 134, 0.34);
}
.syntax--comment {
  color: #999988;
  font-style: italic;
}
.syntax--string {
  color: #D14;
}
.syntax--string .syntax--source,
.syntax--string .syntax--meta.syntax--embedded.syntax--line {
  color: #5A5A5A;
}
.syntax--string .syntax--punctuation.syntax--section.syntax--embedded {
  color: #920B2D;
}
.syntax--string .syntax--punctuation.syntax--section.syntax--embedded .syntax--source {
  color: #920B2D;
}
.syntax--constant.syntax--numeric {
  color: #D14;
}
.syntax--constant.syntax--language {
  color: #606aa1;
}
.syntax--constant.syntax--character,
.syntax--constant.syntax--other {
  color: #606aa1;
}
.syntax--constant.syntax--symbol {
  color: #990073;
}
.syntax--constant.syntax--numeric.syntax--line-number.syntax--find-in-files .syntax--match {
  color: rgba(143, 190, 0, 0.63);
}
.syntax--variable {
  color: #008080;
}
.syntax--variable.syntax--parameter {
  color: #606aa1;
}
.syntax--keyword {
  color: #222;
  font-weight: bold;
}
.syntax--keyword.syntax--unit {
  color: #445588;
}
.syntax--keyword.syntax--special-method {
  color: #0086B3;
}
.syntax--storage {
  color: #222;
}
.syntax--storage.syntax--type {
  color: #222;
}
.syntax--entity.syntax--name.syntax--class {
  text-decoration: underline;
  color: #606aa1;
}
.syntax--entity.syntax--other.syntax--inherited-class {
  text-decoration: underline;
  color: #606aa1;
}
.syntax--entity.syntax--name.syntax--function {
  color: #900;
}
.syntax--entity.syntax--name.syntax--tag {
  color: #008080;
}
.syntax--entity.syntax--other.syntax--attribute-name {
  color: #458;
  font-weight: bold;
}
.syntax--entity.syntax--name.syntax--filename.syntax--find-in-files {
  color: #E6DB74;
}
.syntax--support.syntax--constant,
.syntax--support.syntax--function,
.syntax--support.syntax--type {
  color: #458;
}
.syntax--support.syntax--class {
  color: #008080;
}
.syntax--invalid {
  color: #F8F8F0;
  background-color: #00A8C6;
}
.syntax--invalid.syntax--deprecated {
  color: #F8F8F0;
  background-color: #8FBE00;
}
.syntax--meta.syntax--structure.syntax--dictionary.syntax--json > .syntax--string.syntax--quoted.syntax--double.syntax--json,
.syntax--meta.syntax--structure.syntax--dictionary.syntax--json > .syntax--string.syntax--quoted.syntax--double.syntax--json .syntax--punctuation.syntax--string {
  color: #000080;
}
.syntax--meta.syntax--structure.syntax--dictionary.syntax--value.syntax--json > .syntax--string.syntax--quoted.syntax--double.syntax--json {
  color: #d14;
}
.syntax--meta.syntax--diff,
.syntax--meta.syntax--diff.syntax--header {
  color: #75715E;
}
.syntax--css.syntax--support.syntax--property-name {
  font-weight: bold;
  color: #333;
}
.syntax--css.syntax--constant {
  color: #099;
}
.syntax--source.syntax--gfm {
  color: #444;
}
.syntax--gfm .syntax--markup.syntax--heading {
  color: #111;
}
.syntax--gfm .syntax--link {
  color: #888;
}
.syntax--gfm .syntax--variable.syntax--list {
  color: #888;
}
.syntax--markdown .syntax--paragraph {
  color: #444;
}
.syntax--markdown .syntax--heading {
  color: #111;
}
.syntax--markdown .syntax--link {
  color: #888;
}
.syntax--markdown .syntax--link .syntax--string {
  color: #888;
}
</style>
  </head>
  <body class='markdown-preview' data-use-github-style><h1 id="biological-replicates-practical">Biological replicates practical</h1>
<p>Sandra Cortijo <a href="mailto:sandra.cortijo@slcu.cam.ac.uk">sandra.cortijo@slcu.cam.ac.uk</a><br>Sergio Martinez Cuesta <a href="mailto:sermarcue@gmail.com">sermarcue@gmail.com</a><br>Ashley Sawle <a href="mailto:Ashley.Sawle@cruk.cam.ac.uk">Ashley.Sawle@cruk.cam.ac.uk</a><br>Denis Seyres <a href="mailto:Denis.seyres@bioresource.nihr.ac.uk">Denis.seyres@bioresource.nihr.ac.uk</a></p>
<p>(based on the practical by Myrto Kostadima)</p>
<h2 id="introduction">Introduction</h2>
<p>In this practical, we will use ChIP-seq data for the PAX5 transcription factor, generated on the GM12878 lymphoblastoid cell line. BAM files for the two replicates and a matched control Input file have been downloaded from <a href="http://www.encodeproject.org">http://www.encodeproject.org</a>, named PAX5_BR1TR1_ENCFF614SHU.bam, PAX5_BR2TR1_ENCFF475VQL.bam and Control.bam, respectively.</p>
<h2 id="genome-wide-correlation">Genome-wide correlation</h2>
<p>We will first calculate genome-wide correlation between the read coverage of the two PAX5 ChIP-seq replicates, a basic measure of reproducibility across experiments.</p>
<p>Open the Terminal and go to the folder containing the data.</p>
<pre class="editor-colors lang-bash"><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function"><span class="syntax--support syntax--function">cd</span></span> ~/Desktop/Course_Materials/Practicals/idr</span></div></pre>
<p>The first step towards calculating pairwise correlation between the two biological replicates is to compute read coverages over genomic regions using the BAM files. The analysis can be performed for the entire genome by running the tool <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">multiBamSummary</code> in 'bins' mode. By default the coverage calculation is done for consecutive bins of 10 kilobases long.</p>
<pre class="editor-colors lang-bash"><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">multiBamSummary</span> bins <span class="syntax--entity syntax--other syntax--attribute-name">--bamfiles</span> processed_data/bam/PAX5_BR1TR1_ENCFF614SHU.bam processed_data/bam/PAX5_BR2TR1_ENCFF475VQL.bam <span class="syntax--entity syntax--other syntax--attribute-name">--labels</span> PAX5_BR1TR1 PAX5_BR2TR1 <span class="syntax--entity syntax--other syntax--attribute-name">-out</span> processed_data/coverage/PAX5_read_counts.npz <span class="syntax--entity syntax--other syntax--attribute-name">--outRawCounts</span> processed_data/coverage/PAX5_read_counts.tab</span></div></pre>
<p>Finally we compute the Pearson correlation of these two PAX5 ChIP-seq profiles, using the command <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">plotCorrelation</code>.</p>
<pre class="editor-colors lang-bash"><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">plotCorrelation</span> <span class="syntax--entity syntax--other syntax--attribute-name">-in</span> processed_data/coverage/PAX5_read_counts.npz <span class="syntax--entity syntax--other syntax--attribute-name">--corMethod</span> pearson <span class="syntax--entity syntax--other syntax--attribute-name">--skipZeros</span> <span class="syntax--entity syntax--other syntax--attribute-name">--plotTitle</span> <span class="syntax--string">"Pearson Correlation of Average Reads per Bin"</span> <span class="syntax--entity syntax--other syntax--attribute-name">--whatToPlot</span> scatterplot <span class="syntax--entity syntax--other syntax--attribute-name">-o</span> processed_data/coverage/PAX5_pearson_corr.png <span class="syntax--entity syntax--other syntax--attribute-name">--outFileCorMatrix</span> processed_data/coverage/PAX5_pearson_corr.tab <span class="syntax--entity syntax--other syntax--attribute-name">--removeOutliers</span></span></div></pre>
<h3 id="questions">Questions</h3>
<ul>
<li>What is the correlation between these two samples?</li>
<li>Is this higher or lower than you expected?</li>
</ul>
<h2 id="working-with-two-biological-replicates-chance-statistics">Working with two biological replicates - CHANCE statistics</h2>
<p>CHANCE (CHIP-seq ANalytics and Confidence Estimation) is a standalone package for ChIP-seq quality control and protocol optimization first published in <a href="https://genomebiology.biomedcentral.com/articles/10.1186/gb-2012-13-10-r98">Diaz et al., Genome Biology (2012)</a>. The authors provide a user-friendly graphical software that among other functionalities it estimates the strength and quality of immunoprecipitations (CHANCE is available at <a href="https://github.com/songlab/chance)">https://github.com/songlab/chance)</a>. For the purposes of our practical we will calculate the enrichment of our two PAX5 biological replicates over our Control sample, not using the CHANCE tool itself, but using <a href="https://deeptools.readthedocs.io/en/latest/">deepTools</a>.</p>
<p>First, we need to make sure that all our BAM files, for both replicates and Control samples are indexed. Have a look at the <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">processed_data/bam</code> folder to see if all BAM files have their respective <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">.bam.bai</code> file. If that's not the case, create indexes for all the bam files using <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">samtools index</code>, e.g. <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">samtools index PAX5_BR1TR1_ENCFF614SHU.bam</code>.</p>
<p>Once all the index files have been created, run the following command to calculate the enrichment of the sample PAX5_BR1TR1_ENCFF614SHU over the Control using bin sizes of 1000bp.</p>
<pre class="editor-colors lang-bash"><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">plotFingerprint</span> <span class="syntax--entity syntax--other syntax--attribute-name">-b</span> processed_data/bam/PAX5_BR1TR1_ENCFF614SHU.bam processed_data/bam/Control.bam <span class="syntax--entity syntax--other syntax--attribute-name">--binSize</span> 1000 <span class="syntax--entity syntax--other syntax--attribute-name">--labels</span> PAX5_BR1TR1 Control <span class="syntax--entity syntax--other syntax--attribute-name">--JSDsample</span> processed_data/bam/Control.bam <span class="syntax--entity syntax--other syntax--attribute-name">--outQualityMetrics</span> processed_data/bam/PAX5_BR1TR1_ENCFF614SHU_fingerprint.txt <span class="syntax--entity syntax--other syntax--attribute-name">-plot</span> processed_data/bam/PAX5_BR1TR1_ENCFF614SHU_fingerprint.png</span></div></pre>
<p>For a detailed explanation of the output <em>_fingerprint.txt</em> file format, read <a href="https://deeptools.readthedocs.io/en/latest/content/feature/plotFingerprint_QC_metrics.html">this</a>. Now run the above command for the other replicate as well. Once the tool has finished, open both the PNG files created.</p>
<h3 id="question">Question</h3>
<ul>
<li>Is the enrichment similar for both PAX5 biological replicates? If not, what differences do you observe?</li>
</ul>
<h2 id="working-with-two-biological-replicates-irreproducible-discovery-rate-idr-analysis">Working with two biological replicates - Irreproducible Discovery Rate (IDR) analysis</h2>
<p>Peak calling using MACS2 was performed for both replicates individually and after merging the alignments from both replicates, e.g. using <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">samtools merge</code>, to create a pooled peak set. These datasets are already provided for you in the <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">processed_data/macs2</code> directory, they are labelled <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">PAX5_BR1TR1_peaks.bed</code>, <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">PAX5_BR2TR1_peaks.bed</code> and <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">PAX5_pooled_peaks.bed</code>, respectively.</p>
<p>The narrowPeak format in which the peaks are reported is an ENCODE format, using an extension of the BED format for providing peaks and associated scores and p-values. See <a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format12">this</a> for full information.</p>
<p>The ChIP-seq experiment and alignment may be affected by read mapping artefacts caused by biases in chromatin accessibility and ambiguous alignments. Some of these spurious regions can be removed by filtering out any peaks that overlap a blacklist, believed to contain experiment and cell type independent areas of high artefactual signal. The ENCODE blacklist has been generated by combining regions of known repeats and manually curated genomic regions of ubiquitous open chromatin and input sequence signal. The file can be downloaded from <a href="https://www.encodeproject.org/annotations/ENCSR636HFF/">here</a>.</p>
<p>We can now remove any peaks which overlap with the ENCODE blacklist:</p>
<pre class="editor-colors lang-bash"><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">bedtools</span> intersect <span class="syntax--entity syntax--other syntax--attribute-name">-a</span> processed_data/macs2/PAX5_BR1TR1_peaks.bed <span class="syntax--entity syntax--other syntax--attribute-name">-b</span> annotation/GRCh38_blacklisted_regions.bed <span class="syntax--entity syntax--other syntax--attribute-name">-v</span> <span class="syntax--keyword syntax--operator">&gt;</span> processed_data/macs2/PAX5_BR1TR1_peaks_filtered.bed</span></div><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">bedtools</span> intersect <span class="syntax--entity syntax--other syntax--attribute-name">-a</span> processed_data/macs2/PAX5_BR2TR1_peaks.bed <span class="syntax--entity syntax--other syntax--attribute-name">-b</span> annotation/GRCh38_blacklisted_regions.bed <span class="syntax--entity syntax--other syntax--attribute-name">-v</span> <span class="syntax--keyword syntax--operator">&gt;</span> processed_data/macs2/PAX5_BR2TR1_peaks_filtered.bed</span></div><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">bedtools</span> intersect <span class="syntax--entity syntax--other syntax--attribute-name">-a</span> processed_data/macs2/PAX5_pooled_peaks.bed <span class="syntax--entity syntax--other syntax--attribute-name">-b</span> annotation/GRCh38_blacklisted_regions.bed <span class="syntax--entity syntax--other syntax--attribute-name">-v</span> <span class="syntax--keyword syntax--operator">&gt;</span> processed_data/macs2/PAX5_pooled_peaks_filtered.bed</span></div></pre>
<p>Next, we need to sort peaks in the files by significance (p-value). For <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">MACS2</code>, the p-value works best for ranking using the IDR procedure. See <a href="https://sites.google.com/site/ anshulkundaje/projects/idr#TOC-Peak-callers-tested-with-IDR">this</a> for the best ranking measures to use with other peak callers.</p>
<p>Here, we sort by p-value and then use the 100,000 most significant peaks:</p>
<pre class="editor-colors lang-bash"><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">sort</span> <span class="syntax--entity syntax--other syntax--attribute-name">-k</span> 8fr processed_data/macs2/PAX5_BR1TR1_peaks_filtered.bed <span class="syntax--keyword syntax--operator">|</span> <span class="syntax--entity syntax--name syntax--function">head</span> <span class="syntax--entity syntax--other syntax--attribute-name">-n</span> 100000 <span class="syntax--keyword syntax--operator">&gt;</span> processed_data/macs2/PAX5_BR1TR1_top100000_peaks.bed</span></div><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">sort</span> <span class="syntax--entity syntax--other syntax--attribute-name">-k</span> 8fr processed_data/macs2/PAX5_BR2TR1_peaks_filtered.bed <span class="syntax--keyword syntax--operator">|</span> <span class="syntax--entity syntax--name syntax--function">head</span> <span class="syntax--entity syntax--other syntax--attribute-name">-n</span> 100000 <span class="syntax--keyword syntax--operator">&gt;</span> processed_data/macs2/PAX5_BR2TR1_top100000_peaks.bed</span></div><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">sort</span> <span class="syntax--entity syntax--other syntax--attribute-name">-k</span> 8fr processed_data/macs2/PAX5_pooled_peaks_filtered.bed <span class="syntax--keyword syntax--operator">|</span> <span class="syntax--entity syntax--name syntax--function">head</span> <span class="syntax--entity syntax--other syntax--attribute-name">-n</span> 100000 <span class="syntax--keyword syntax--operator">&gt;</span> processed_data/macs2/PAX5_pooled_top100000_peaks.bed</span></div></pre>
<p>Finally, we can perform the IDR analysis on the peaks called in the two technical replicates, using the peak list from the merged replicates</p>
<pre class="editor-colors lang-bash"><div class="line"><span class="syntax--source syntax--shell"><span class="syntax--entity syntax--name syntax--function">idr</span> <span class="syntax--entity syntax--other syntax--attribute-name">--samples</span> processed_data/macs2/PAX5_BR1TR1_top100000_peaks.bed processed_data/macs2/PAX5_BR2TR1_top100000_peaks.bed <span class="syntax--entity syntax--other syntax--attribute-name">--peak-list</span> processed_data/macs2/PAX5_pooled_top100000_peaks.bed <span class="syntax--entity syntax--other syntax--attribute-name">--idr-threshold</span> 0.05 <span class="syntax--entity syntax--other syntax--attribute-name">--output-file</span> idr/PAX5_replicates_idr.txt <span class="syntax--entity syntax--other syntax--attribute-name">--plot</span></span></div></pre>
<h3 id="questions-understanding-the-idr-output-plot">Questions: understanding the IDR output plot</h3>
<ul>
<li>How reproducible are the peaks called in these two biological replicates for PAX5 binding in the GM12878 cell line?</li>
<li>What factors could lower the reproducibility between the two ChIP-seq experiments?</li>
</ul>
<p>CONGRATULATIONS! :) You’ve made it to the end of the practical.</p>
<h2 id="bonus-exercise-1">Bonus Exercise 1</h2>
<p>The IDR statistics can also be used to flag data sets with low reproducibility. This may be due to one of the two replicates being of lower ChIP enrichment, hence having a high signal-noise ratio. In this case, the standard IDR protocol would record few reproducible peaks, despite one replicate having high information content.</p>
<p>ENCODE has developed a rescue strategy in this case by using pseudo-replicates. These pseudo-replicates are generated by pooling all the reads, and then randomly splitting them into two files. These pseudo-replicates do not represent true biological or technical replicates, but attempt to model the stochastic noise in the sampling of sequenced reads from a population of fragments.</p>
<p>The pseudo-replicates analysis uses a lower IDR threshold than with biological/technical replicates, typically 0.0025, due to the reduced noise. Check the <code style="font-family: Menlo, Consolas, &quot;DejaVu Sans Mono&quot;, monospace;">idr</code> help in the Terminal and modifiy the command above to run the IDR analysis for pseudo-replicates of the GM12878 PAX5 ChIP-seq data.</p>
<h3 id="question">Question</h3>
<ul>
<li>Does the IDR method select more peaks using the original technical replicates or the pseudo-replicates approach?</li>
</ul></body>
</html>
